onnxruntime/core/session/environment.cc

```diff
@@ -312,6 +322,11 @@ Internal copy node
     // fire off startup telemetry (this call is idempotent)
     const Env& env = Env::Default();
     env.GetTelemetryProvider().LogProcessInfo();
+
+    // register internal EPs for autoep selection
+    // TODO: ??? Is there any reason not to do this like an EP allocates a large chunk of memory when created?
+    //       If that is the case the user could register by name with no library path to do registration manually.
```

Comment:
I think the CPU EP is fine. Any concerns about always creating an instance of the WebGPU or DML EPs if they're included in the build?

Reply:
for static EP's right? seems ok to me. we can check how much mem those EP's allocate upon creation

Reply:
yeah - static EPs only. can refine how they behave in a separate PR if needed.